src/test/scala/com/microsoft/hyperspace/index/IndexStatisticsTest.scala

```diff
@@ -0,0 +1,154 @@
+  }
+
+  override def afterAll(): Unit = {
+    FileUtils.delete(new Path(dataPath), true)
+    super.afterAll()
+  }
+
+  after {
+    FileUtils.delete(systemPath)
+  }
+
+  test("getIndexStats() on a fresh index returns correct result.") {
+    Seq(true, false).foreach { enableLineage =>
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> enableLineage.toString) {
+        withIndex(indexConfig.indexName) {
+          hyperspace.createIndex(dataDF, indexConfig)
+          validateIndexStats(
+            indexConfig.indexName,
+            Seq(IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX + "=0"))
+        }
+      }
+    }
+  }
+
+  test(
+    "getIndexStats() on an index refreshed in incremental or quick mode returns correct result.") {
+    Seq("incremental", "quick").foreach { mode =>
+      withTempPathAsString { testPath =>
+        withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+          withIndex(indexConfig.indexName) {
+            SampleData.save(spark, testPath, Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+            val df = spark.read.parquet(testPath)
+            hyperspace.createIndex(df, indexConfig)
+
+            // Modify source content.
+            import spark.implicits._
+            TestUtils.deleteFiles(testPath, "*parquet", 1).head
+            SampleData.testData
+              .take(3)
+              .toDF(dataColumns: _*)
+              .write
+              .mode("append")
+              .parquet(testPath)
+
+            hyperspace.refreshIndex(indexConfig.indexName, mode)
+
+            val version = if (mode.equals("quick")) 0 else 1
+            validateIndexStats(
+              indexConfig.indexName,
+              Seq(IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX + s"=$version"))
+          }
+        }
+      }
+    }
+  }
+
+  test("getIndexStats() on an index with multiple active versions returns correct result.") {
+    withTempPathAsString { testPath =>
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+        withIndex(indexConfig.indexName) {
+          SampleData.save(spark, testPath, Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+          val df = spark.read.parquet(testPath)
+          hyperspace.createIndex(df, indexConfig)
+
+          import spark.implicits._
+          var expectedVersions = Seq(IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX + "=0")
+          for (i <- 1 to 2) {
+            SampleData.testData
+              .take(3)
+              .toDF(dataColumns: _*)
+              .write
+              .mode("append")
+              .parquet(testPath)
+
+            hyperspace.refreshIndex(indexConfig.indexName, "incremental")
+            expectedVersions :+= IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX + s"=$i"
+          }
+          validateIndexStats(indexConfig.indexName, expectedVersions)
+        }
+      }
+    }
+  }
+
+  private def validateIndexStats(indexName: String, expectedIndexPaths: Seq[String]): Unit = {
```

Comment:
It should be `expectedIndexVersions`? In this case, can we just pass `Seq[Int]` to simplify?